### PR TITLE
Adds a confirmation alert

### LINKED ISSUE #1930: Add confirmation messages when submitting a correction request or making a change
Show some confirmation to the user that the action was completed successfully. This could be as simple as a toast message on the Circulars archive page.

### PR DESCRIPTION
**Pull request by dakota002**
Closes #1930

### CODE REVIEW

**Review by Courey — APPROVED** [review]

**Review by lpsinger — CHANGES_REQUESTED** [review]
Rather than returning a new boolean, how about returning `intent` so that you can, if need be, display a different confirmation message depending on the operation?

File: app/routes/_gcn.circulars._archive._index/route.tsx [hunk 1]
```diff
@@ -115,24 +118,27 @@ export async function action({ request }: ActionFunctionArgs) {
...
       break
     case 'new':
-      result = await put({ ...props, submittedHow: 'web' }, user)
+      newItem = await put({ ...props, submittedHow: 'web' }, user)
       break
     default:
       break
   }
-  return result
+  return { newItem, showSuccess }
```
Comment (lpsinger) [inline app/routes/_gcn.circulars._archive._index/route.tsx:129]: Rather than call it `result`, how about `newCircular`?

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/_gcn.circulars._archive._index/route.tsx (hunk 1 above)
Comment (dakota002) [reply app/routes/_gcn.circulars._archive._index/route.tsx:129]: In the action? Or in the component function? In the latter that will lead to it looking like 

```ts
const newCircular= useActionData<typeof action>() 
// const {newItem, intent} = useActionData<typeof action>() throws 
// errors about newItem and intent not being defined
...
const allItems = [
    ...(newCircular?.newItem ? [newCircular.newItem] : []),
    ...(items || []),
  ]
```

Comment (lpsinger) [reply app/routes/_gcn.circulars._archive._index/route.tsx:129]:
```suggestion
  return { newCircular, intent }
```

**Review by lpsinger — CHANGES_REQUESTED** [review]

**Review by lpsinger — APPROVED** [review]

**Comment by dakota002** [comment]
@lpsinger What are your thoughts on this? Went with your wording on the sentence and moved the 'Request Submitted' text to the heading of the alert

image

**Comment by lpsinger** [comment]
Yup, looks good!

**Comment by dakota002** [comment]
Nice, new PR out for it ^

**Pull request merged**